### ISSUE TITLE
write back new meteor config fails: value must be a string

### ISSUE THREAD

**Issue by fkohrt** [n0emis/ansible-role-bigbluebutton#40]
I have set up a playbook as below, but it fails with the following message:
```
TASK [n0emis.bigbluebutton : write back new meteor config] *********************
fatal: [<hostname>]: FAILED! => {"msg": "Unexpected templating type error occurred on ({{ meteor  | to_nice_yaml }}): value must be a string"}
```

my `playbook.yml`:

```yaml
- hosts: webservers
  remote_user: <username>
  become: yes
  roles:
     - role: n0emis.bigbluebutton
       bbb_hostname: <hostname>
       bbb_letsencrypt_email: <email>
       bbb_coturn_secret: <coturn secret>
       bbb_greenlight_secret: <greenlight secret>
       bbb_greenlight_db_password: <greenlight db password>
       bbb_greenlight_default_registration: open
       bbb_disable_recordings: yes
       bbb_meteor:
         public:
           app:
             autoJoin: true
             skipCheck: true
       bbb_mute_on_start: yes
       bbb_freeswitch_ipv6: false
       bbb_greenlight_smtp:
         server: <smtp server>
         port: <smtp port>
         domain: <smtp domain>
         username: <smtp username>
         password: <smtp password>
         auth: plain
         starttls_auto: true
         sender: <smtp sender>
```

Is there something wrong with the playbook? Or is it there an issue with the role? Removing the `bbb_meteor` part completely still leads to that error...

**Comment by n0emis** [comment]
Hmm, I can't reproduce this. You can find the Variables I tested with under https://git.n0emis.eu/n0emis/home-automation/src/commit/792ba23b2a3f559bc4e9c5674a69ccf6ef996514/group_vars/bbb.yml#L26 . 

Which version of ansible are you using?

**Comment by fkohrt** [comment]
Sorry, been busy the last days; I had needed the server on short notice and therefore turned to a more manual solution, but I will look again if I can reproduce this (the system was a standard Azure VM).

**Comment by n0emis** [comment]
Coud you send me the values of the `meteor` variables before and after combining? You can do this by insterting 
```yaml
- name: extract config of meteor
  set_fact:
    meteor: "{{ r_meteor['content'] | b64decode | from_yaml }}"
  
- debug:
    msg: "{{ meteor }}"
    
- name: combine meteor config
  set_fact:
    meteor: "{{ meteor | combine(bbb_meteor, recursive=true) }}"
  
- debug:
    msg: "{{ meteor }}"
    
- name: write back new meteor config
  notify: restart bigbluebutton
  copy:
    content: '{{ meteor  | to_nice_yaml }}'
    dest: /usr/share/meteor/bundle/programs/server/assets/app/config/settings.yml
```
into the `tasks/config.yml`.

**Comment by fkohrt** [comment]
@n0emis Here you go, the error still occurs. I'm grateful for any advice!
```
TASK [n0emis.bigbluebutton : debug] ********************************************
ok: [<hostname>] => {
    "msg": {
        "private": {
            "app": {
                "host": "127.0.0.1",
                "loadSlidesFromHttpAlways": false,
                "localesUrl": "/locales",
                "pencilChunkLength": 100,
                "port": 3000
            },
            "etherpad": {
                "apikey": "Doph3cPHkgEUsGczGer8w6lo9lZt1PBpzUchgVgH7EysJ3rq1UUEzLb813Bf7j",
                "host": "127.0.0.1",
                "port": 9001,
                "version": "1.2.13"
            },
            "memoryMonitoring": {
                "heapdump": {
                    "enabled": false
                },
                "leak": {
                    "enabled": false
                },
                "stat": {
                    "enabled": false
                }
            },
            "minBrowserVersions": [
                {
                    "browser": "chrome",
                    "version": 59
                },
                {
                    "browser": "chromeMobileIOS",
                    "version": "Infinity"
                },
                {
                    "browser": "firefox",
                    "version": 52
                },
                {
                    "browser": "firefoxMobile",
                    "version": 52
                },
                {
                    "browser": "edge",
                    "version": 17
                },
                {
                    "browser": "ie",
                    "version": "Infinity"
                },
                {
                    "browser": "safari",
                    "version": [
                        11,
                        1
                    ]
                },
                {
                    "browser": "mobileSafari",
                    "version": [
                        11,
                        1
                    ]
                },
                {
                    "browser": "opera",
                    "version": 46
                },
                {
                    "browser": "electron",
                    "version": [
                        0,
                        36
                    ]
                }
            ],
            "redis": {
                "async": [
                    "from-akka-apps-wb-redis-channel"
                ],
                "channels": {
                    "toAkkaApps": "to-akka-apps-redis-channel",
                    "toThirdParty": "to-third-party-redis-channel"
                },
                "debug": false,
                "host": "127.0.0.1",
                "ignored": [
                    "CheckAlivePongSysMsg",
                    "DoLatencyTracerMsg"
                ],
                "password": null,
                "port": "6379",
                "subscribeTo": [
                    "to-html5-redis-channel",
                    "from-akka-apps-*",
                    "from-third-party-redis-channel",
                    "from-etherpad-redis-channel"
                ],
                "timeout": 5000
            },
            "serverLog": {
                "level": "info",
                "streamerLog": false
            }
        },
        "public": {
            "allowOutsideCommands": {
                "toggleRecording": false,
                "toggleSelfVoice": false
            },
            "app": {
                "allowFullscreen": true,
                "allowLogout": true,
                "allowUserLookup": false,
                "appName": "BigBlueButton HTML5 Client",
                "askForFeedbackOnLogout": false,
                "audioChatNotification": false,
                "autoJoin": true,
                "basename": "/html5client",
                "bbbServerVersion": 2.2,
                "branding": {
                    "displayBrandingArea": false
                },
                "cdn": "",
                "clientTitle": "BigBlueButton",
                "connectionTimeout": 60000,
                "copyright": "©2019 BigBlueButton Inc.",
                "defaultSettings": {
                    "application": {
                        "animations": true,
                        "chatAudioAlerts": false,
                        "chatPushAlerts": false,
                        "fallbackLocale": "en",
                        "overrideLocale": null,
                        "userJoinAudioAlerts": false,
                        "userJoinPushAlerts": false
                    },
                    "audio": {
                        "inputDeviceId": "undefined",
                        "outputDeviceId": "undefined"
                    },
                    "dataSaving": {
                        "viewParticipantsWebcams": true,
                        "viewScreenshare": true
                    }
                },
                "desktopFontSize": "14px",
                "effectiveConnection": [
                    "critical",
                    "danger",
                    "warning"
                ],
                "enableLimitOfViewersInWebcam": false,
                "enableNetworkInformation": false,
                "enableTalkingIndicator": true,
                "forceListenOnly": false,
                "helpLink": "https://bigbluebutton.org/html5/",
                "html5ClientBuild": 934,
                "ipv4FallbackDomain": "",
                "listenOnlyMode": false,
                "lockOnJoin": true,
                "mobileFontSize": "16px",
                "shortcuts": {
                    "closePrivateChat": {
                        "accesskey": "G",
                        "descId": "closePrivateChat"
                    },
                    "hidePrivateChat": {
                        "accesskey": "H",
                        "descId": "hidePrivateChat"
                    },
                    "joinAudio": {
                        "accesskey": "J",
                        "descId": "joinAudio"
                    },
                    "leaveAudio": {
                        "accesskey": "L",
                        "descId": "leaveAudio"
                    },
                    "openActions": {
                        "accesskey": "A",
                        "descId": "openActions"
                    },
                    "openOptions": {
                        "accesskey": "O",
                        "descId": "openOptions"
                    },
                    "toggleMute": {
                        "accesskey": "M",
                        "descId": "toggleMute"
                    },
                    "togglePublicChat": {
                        "accesskey": "P",
                        "descId": "togglePublicChat"
                    },
                    "toggleUserList": {
                        "accesskey": "U",
                        "descId": "toggleUserList"
                    }
                },
                "showHelpButton": true,
                "skipCheck": true,
                "viewersInWebcam": 8
            },
            "captions": {
                "backgroundColor": "#000000",
                "enableDictation": false,
                "enabled": true,
                "fontColor": "#FFFFFF",
                "fontFamily": "Calibri",
                "fontSize": "24px",
                "lines": 2,
                "takeOwnership": true,
                "time": 5000
            },
            "chat": {
                "enabled": true,
                "grouping_messages_window": 10000,
                "max_message_length": 5000,
                "min_message_length": 1,
                "public_group_id": "MAIN-PUBLIC-GROUP-CHAT",
                "public_id": "public",
                "public_userid": "public_chat_userid",
                "public_username": "public_chat_username",
                "startClosed": false,
                "storage_key": "UNREAD_CHATS",
                "system_messages_keys": {
                    "chat_clear": "PUBLIC_CHAT_CLEAR"
                },
                "system_userid": "SYSTEM_MESSAGE",
                "system_username": "SYSTEM_MESSAGE",
                "type_private": "PRIVATE_ACCESS",
                "type_public": "PUBLIC_ACCESS",
                "type_system": "SYSTEM_MESSAGE",
                "typingIndicator": {
                    "enabled": true
                }
            },
            "clientLog": {
                "console": {
                    "enabled": true,
                    "level": "debug"
                },
                "external": {
                    "enabled": false,
                    "flushOnClose": true,
                    "level": "info",
                    "logTag": "",
                    "method": "POST",
                    "throttleInterval": 400,
                    "url": "https://LOG_HOST/html5Log"
                },
                "server": {
                    "enabled": true,
                    "level": "info"
                }
            },
            "externalVideoPlayer": {
                "enabled": true
            },
            "kurento": {
                "autoShareWebcam": false,
                "cameraProfiles": [
                    {
                        "bitrate": 50,
                        "default": false,
                        "id": "low",
                        "name": "Low quality"
                    },
                    {
                        "bitrate": 100,
                        "default": false,
                        "id": "medium",
                        "name": "Medium quality"
                    },
                    {
                        "bitrate": 200,
                        "default": false,
                        "id": "high",
                        "name": "High quality"
                    },
                    {
                        "bitrate": 300,
                        "default": true,
                        "id": "hd",
                        "name": "High definition"
                    }
                ],
                "chromeDefaultExtensionKey": "akgoaoikmbmhcopjgakkcepdgdgkjfbc",
                "chromeDefaultExtensionLink": "https://chrome.google.com/webstore/detail/bigbluebutton-screenshare/akgoaoikmbmhcopjgakkcepdgdgkjfbc",
                "chromeExtensionKey": "KEY",
                "chromeExtensionLink": "LINK",
                "enableListenOnly": true,
                "enableScreensharing": true,
                "enableVideo": true,
                "enableVideoMenu": true,
                "enableVideoStats": false,
                "screenshare": {
                    "chromeScreenshareSources": [
                        "window",
                        "screen"
                    ],
                    "constraints": {
                        "audio": false,
                        "video": {
                            "frameRate": {
                                "ideal": 5,
                                "max": 10
                            },
                            "height": {
                                "max": 1600
                            },
                            "width": {
                                "max": 2560
                            }
                        }
                    },
                    "firefoxScreenshareSource": "window"
                },
                "skipVideoPreview": false,
                "wsUrl": "wss://<hostname>/bbb-webrtc-sfu"
            },
            "layout": {
                "autoSwapLayout": false,
                "hidePresentation": false
            },
            "media": {
                "cacheStunTurnServers": true,
                "callHangupMaximumRetries": 10,
                "callHangupTimeout": 2000,
                "callTransferTimeout": 5000,
                "echoTestNumber": "echo",
                "fallbackStunServer": "",
                "mediaTag": "#remote-media",
                "relayOnlyOnReconnect": false,
                "stunTurnServersFetchAddress": "/bigbluebutton/api/stuns"
            },
            "networkMonitoring": {
                "enableNetworkMonitoring": false,
                "packetLostThreshold": 10
            },
            "note": {
                "config": {
                    "noColors": true,
                    "rtl": false,
                    "showChat": false,
                    "showControls": true,
                    "showLineNumbers": false
                },
                "enabled": true,
                "url": "https://<hostname>/pad"
            },
            "pingPong": {
                "clearUsersInSeconds": 180,
                "pongTimeInSeconds": 15
            },
            "poll": {
                "enabled": true,
                "max_custom": 5
            },
            "presentation": {
                "defaultPresentationFile": "default.pdf",
                "panZoomThrottle": 32,
                "restoreOnUpdate": false,
                "uploadEndpoint": "/bigbluebutton/presentation/upload",
                "uploadSizeMax": 50000000,
                "uploadSizeMin": 0,
                "uploadValidMimeTypes": [
                    {
                        "extension": ".pdf",
                        "mime": "application/pdf"
                    },
                    {
                        "extension": ".doc",
                        "mime": "application/msword"
                    },
                    {
                        "extension": ".docx",
                        "mime": "application/vnd.openxmlformats-officedocument.wordprocessingml.document"
                    },
                    {
                        "extension": ".xls",
                        "mime": "application/vnd.ms-excel"
                    },
                    {
                        "extension": ".xlsx",
                        "mime": "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
                    },
                    {
                        "extension": ".ppt",
                        "mime": "application/vnd.ms-powerpoint"
                    },
                    {
                        "extension": ".pptx",
                        "mime": "application/vnd.openxmlformats-officedocument.presentationml.presentation"
                    },
                    {
                        "extension": ".txt",
                        "mime": "text/plain"
                    },
                    {
                        "extension": ".rtf",
                        "mime": "application/rtf"
                    },
                    {
                        "extension": ".odt",
                        "mime": "application/vnd.oasis.opendocument.text"
                    },
                    {
                        "extension": ".ods",
                        "mime": "application/vnd.oasis.opendocument.spreadsheet"
                    },
                    {
                        "extension": ".odp",
                        "mime": "application/vnd.oasis.opendocument.presentation"
                    },
                    {
                        "extension": ".odg",
                        "mime": "application/vnd.oasis.opendocument.graphics"
                    },
                    {
                        "extension": ".odc",
                        "mime": "application/vnd.oasis.opendocument.chart"
                    },
                    {
                        "extension": ".odi",
                        "mime": "application/vnd.oasis.opendocument.image"
                    },
                    {
                        "extension": ".jpg",
                        "mime": "image/jpeg"
                    },
                    {
                        "extension": ".jpeg",
                        "mime": "image/jpeg"
                    },
                    {
                        "extension": ".png",
                        "mime": "image/png"
                    }
                ]
            },
            "user": {
                "role_moderator": "MODERATOR",
                "role_viewer": "VIEWER"
            },
            "whiteboard": {
                "annotations": {
                    "status": {
                        "end": "DRAW_END",
                        "start": "DRAW_START",
                        "update": "DRAW_UPDATE"
                    }
                },
                "toolbar": {
                    "colors": [
                        {
                            "label": "black",
                            "value": "#000000"
                        },
                        {
                            "label": "white",
                            "value": "#ffffff"
                        },
                        {
                            "label": "red",
                            "value": "#ff0000"
                        },
                        {
                            "label": "orange",
                            "value": "#ff8800"
                        },
                        {
                            "label": "eletricLime",
                            "value": "#ccff00"
                        },
                        {
                            "label": "Lime",
                            "value": "#00ff00"
                        },
                        {
                            "label": "Cyan",
                            "value": "#00ffff"
                        },
                        {
                            "label": "dodgerBlue",
                            "value": "#0088ff"
                        },
                        {
                            "label": "blue",
                            "value": "#0000ff"
                        },
                        {
                            "label": "violet",
                            "value": "#8800ff"
                        },
                        {
                            "label": "magenta",
                            "value": "#ff00ff"
                        },
                        {
                            "label": "silver",
                            "value": "#c0c0c0"
                        }
                    ],
                    "font_sizes": [
                        {
                            "value": 36
                        },
                        {
                            "value": 32
                        },
                        {
                            "value": 28
                        },
                        {
                            "value": 24
                        },
                        {
                            "value": 20
                        },
                        {
                            "value": 16
                        }
                    ],
                    "multiUserPenOnly": false,
                    "multiUserTools": [
                        "text",
                        "line",
                        "ellipse",
                        "triangle",
                        "rectangle",
                        "pencil",
                        "hand"
                    ],
                    "presenterTools": [
                        "text",
                        "line",
                        "ellipse",
                        "triangle",
                        "rectangle",
                        "pencil",
                        "hand"
                    ],
                    "thickness": [
                        {
                            "value": 14
                        },
                        {
                            "value": 12
                        },
                        {
                            "value": 10
                        },
                        {
                            "value": 8
                        },
                        {
                            "value": 6
                        },
                        {
                            "value": 4
                        },
                        {
                            "value": 2
                        },
                        {
                            "value": 1
                        }
                    ],
                    "tools": [
                        {
                            "icon": "text_tool",
                            "value": "text"
                        },
                        {
                            "icon": "line_tool",
                            "value": "line"
                        },
                        {
                            "icon": "circle_tool",
                            "value": "ellipse"
                        },
                        {
                            "icon": "triangle_tool",
                            "value": "triangle"
                        },
                        {
                            "icon": "rectangle_tool",
                            "value": "rectangle"
                        },
                        {
                            "icon": "pen_tool",
                            "value": "pencil"
                        },
                        {
                            "icon": "hand",
                            "value": "hand"
                        }
                    ]
                }
            }
        }
    }
}

TASK [n0emis.bigbluebutton : combine meteor config] ****************************
ok: [<hostname>]

TASK [n0emis.bigbluebutton : debug] ********************************************
ok: [<hostname>] => {
    "msg": {
        "private": {
            "app": {
                "host": "127.0.0.1",
                "loadSlidesFromHttpAlways": false,
                "localesUrl": "/locales",
                "pencilChunkLength": 100,
                "port": 3000
            },
            "etherpad": {
                "apikey": "Doph3cPHkgEUsGczGer8w6lo9lZt1PBpzUchgVgH7EysJ3rq1UUEzLb813Bf7j",
                "host": "127.0.0.1",
                "port": 9001,
                "version": "1.2.13"
            },
            "memoryMonitoring": {
                "heapdump": {
                    "enabled": false
                },
                "leak": {
                    "enabled": false
                },
                "stat": {
                    "enabled": false
                }
            },
            "minBrowserVersions": [
                {
                    "browser": "chrome",
                    "version": 59
                },
                {
                    "browser": "chromeMobileIOS",
                    "version": "Infinity"
                },
                {
                    "browser": "firefox",
                    "version": 52
                },
                {
                    "browser": "firefoxMobile",
                    "version": 52
                },
                {
                    "browser": "edge",
                    "version": 17
                },
                {
                    "browser": "ie",
                    "version": "Infinity"
                },
                {
                    "browser": "safari",
                    "version": [
                        11,
                        1
                    ]
                },
                {
                    "browser": "mobileSafari",
                    "version": [
                        11,
                        1
                    ]
                },
                {
                    "browser": "opera",
                    "version": 46
                },
                {
                    "browser": "electron",
                    "version": [
                        0,
                        36
                    ]
                }
            ],
            "redis": {
                "async": [
                    "from-akka-apps-wb-redis-channel"
                ],
                "channels": {
                    "toAkkaApps": "to-akka-apps-redis-channel",
                    "toThirdParty": "to-third-party-redis-channel"
                },
                "debug": false,
                "host": "127.0.0.1",
                "ignored": [
                    "CheckAlivePongSysMsg",
                    "DoLatencyTracerMsg"
                ],
                "password": null,
                "port": "6379",
                "subscribeTo": [
                    "to-html5-redis-channel",
                    "from-akka-apps-*",
                    "from-third-party-redis-channel",
                    "from-etherpad-redis-channel"
                ],
                "timeout": 5000
            },
            "serverLog": {
                "level": "info",
                "streamerLog": false
            }
        },
        "public": {
            "allowOutsideCommands": {
                "toggleRecording": false,
                "toggleSelfVoice": false
            },
            "app": {
                "allowFullscreen": true,
                "allowLogout": true,
                "allowUserLookup": false,
                "appName": "BigBlueButton HTML5 Client",
                "askForFeedbackOnLogout": false,
                "audioChatNotification": false,
                "autoJoin": true,
                "basename": "/html5client",
                "bbbServerVersion": 2.2,
                "branding": {
                    "displayBrandingArea": false
                },
                "cdn": "",
                "clientTitle": "BigBlueButton",
                "connectionTimeout": 60000,
                "copyright": "©2019 BigBlueButton Inc.",
                "defaultSettings": {
                    "application": {
                        "animations": true,
                        "chatAudioAlerts": false,
                        "chatPushAlerts": false,
                        "fallbackLocale": "en",
                        "overrideLocale": null,
                        "userJoinAudioAlerts": false,
                        "userJoinPushAlerts": false
                    },
                    "audio": {
                        "inputDeviceId": "undefined",
                        "outputDeviceId": "undefined"
                    },
                    "dataSaving": {
                        "viewParticipantsWebcams": true,
                        "viewScreenshare": true
                    }
                },
                "desktopFontSize": "14px",
                "effectiveConnection": [
                    "critical",
                    "danger",
                    "warning"
                ],
                "enableLimitOfViewersInWebcam": false,
                "enableNetworkInformation": false,
                "enableTalkingIndicator": true,
                "forceListenOnly": false,
                "helpLink": "https://bigbluebutton.org/html5/",
                "html5ClientBuild": 934,
                "ipv4FallbackDomain": "",
                "listenOnlyMode": false,
                "lockOnJoin": true,
                "mobileFontSize": "16px",
                "shortcuts": {
                    "closePrivateChat": {
                        "accesskey": "G",
                        "descId": "closePrivateChat"
                    },
                    "hidePrivateChat": {
                        "accesskey": "H",
                        "descId": "hidePrivateChat"
                    },
                    "joinAudio": {
                        "accesskey": "J",
                        "descId": "joinAudio"
                    },
                    "leaveAudio": {
                        "accesskey": "L",
                        "descId": "leaveAudio"
                    },
                    "openActions": {
                        "accesskey": "A",
                        "descId": "openActions"
                    },
                    "openOptions": {
                        "accesskey": "O",
                        "descId": "openOptions"
                    },
                    "toggleMute": {
                        "accesskey": "M",
                        "descId": "toggleMute"
                    },
                    "togglePublicChat": {
                        "accesskey": "P",
                        "descId": "togglePublicChat"
                    },
                    "toggleUserList": {
                        "accesskey": "U",
                        "descId": "toggleUserList"
                    }
                },
                "showHelpButton": true,
                "skipCheck": true,
                "viewersInWebcam": 8
            },
            "captions": {
                "backgroundColor": "#000000",
                "enableDictation": false,
                "enabled": true,
                "fontColor": "#FFFFFF",
                "fontFamily": "Calibri",
                "fontSize": "24px",
                "lines": 2,
                "takeOwnership": true,
                "time": 5000
            },
            "chat": {
                "enabled": true,
                "grouping_messages_window": 10000,
                "max_message_length": 5000,
                "min_message_length": 1,
                "public_group_id": "MAIN-PUBLIC-GROUP-CHAT",
                "public_id": "public",
                "public_userid": "public_chat_userid",
                "public_username": "public_chat_username",
                "startClosed": false,
                "storage_key": "UNREAD_CHATS",
                "system_messages_keys": {
                    "chat_clear": "PUBLIC_CHAT_CLEAR"
                },
                "system_userid": "SYSTEM_MESSAGE",
                "system_username": "SYSTEM_MESSAGE",
                "type_private": "PRIVATE_ACCESS",
                "type_public": "PUBLIC_ACCESS",
                "type_system": "SYSTEM_MESSAGE",
                "typingIndicator": {
                    "enabled": true
                }
            },
            "clientLog": {
                "console": {
                    "enabled": true,
                    "level": "debug"
                },
                "external": {
                    "enabled": false,
                    "flushOnClose": true,
                    "level": "info",
                    "logTag": "",
                    "method": "POST",
                    "throttleInterval": 400,
                    "url": "https://LOG_HOST/html5Log"
                },
                "server": {
                    "enabled": true,
                    "level": "info"
                }
            },
            "externalVideoPlayer": {
                "enabled": true
            },
            "kurento": {
                "autoShareWebcam": false,
                "cameraProfiles": [
                    {
                        "bitrate": 50,
                        "default": false,
                        "id": "low",
                        "name": "Low quality"
                    },
                    {
                        "bitrate": 100,
                        "default": false,
                        "id": "medium",
                        "name": "Medium quality"
                    },
                    {
                        "bitrate": 200,
                        "default": false,
                        "id": "high",
                        "name": "High quality"
                    },
                    {
                        "bitrate": 300,
                        "default": true,
                        "id": "hd",
                        "name": "High definition"
                    }
                ],
                "chromeDefaultExtensionKey": "akgoaoikmbmhcopjgakkcepdgdgkjfbc",
                "chromeDefaultExtensionLink": "https://chrome.google.com/webstore/detail/bigbluebutton-screenshare/akgoaoikmbmhcopjgakkcepdgdgkjfbc",
                "chromeExtensionKey": "KEY",
                "chromeExtensionLink": "LINK",
                "enableListenOnly": true,
                "enableScreensharing": true,
                "enableVideo": true,
                "enableVideoMenu": true,
                "enableVideoStats": false,
                "screenshare": {
                    "chromeScreenshareSources": [
                        "window",
                        "screen"
                    ],
                    "constraints": {
                        "audio": false,
                        "video": {
                            "frameRate": {
                                "ideal": 5,
                                "max": 10
                            },
                            "height": {
                                "max": 1600
                            },
                            "width": {
                                "max": 2560
                            }
                        }
                    },
                    "firefoxScreenshareSource": "window"
                },
                "skipVideoPreview": false,
                "wsUrl": "wss://<hostname>/bbb-webrtc-sfu"
            },
            "layout": {
                "autoSwapLayout": false,
                "hidePresentation": false
            },
            "media": {
                "cacheStunTurnServers": true,
                "callHangupMaximumRetries": 10,
                "callHangupTimeout": 2000,
                "callTransferTimeout": 5000,
                "echoTestNumber": "echo",
                "fallbackStunServer": "",
                "mediaTag": "#remote-media",
                "relayOnlyOnReconnect": false,
                "stunTurnServersFetchAddress": "/bigbluebutton/api/stuns"
            },
            "networkMonitoring": {
                "enableNetworkMonitoring": false,
                "packetLostThreshold": 10
            },
            "note": {
                "config": {
                    "noColors": true,
                    "rtl": false,
                    "showChat": false,
                    "showControls": true,
                    "showLineNumbers": false
                },
                "enabled": true,
                "url": "https://<hostname>/pad"
            },
            "pingPong": {
                "clearUsersInSeconds": 180,
                "pongTimeInSeconds": 15
            },
            "poll": {
                "enabled": true,
                "max_custom": 5
            },
            "presentation": {
                "defaultPresentationFile": "default.pdf",
                "panZoomThrottle": 32,
                "restoreOnUpdate": false,
                "uploadEndpoint": "/bigbluebutton/presentation/upload",
                "uploadSizeMax": 50000000,
                "uploadSizeMin": 0,
                "uploadValidMimeTypes": [
                    {
                        "extension": ".pdf",
                        "mime": "application/pdf"
                    },
                    {
                        "extension": ".doc",
                        "mime": "application/msword"
                    },
                    {
                        "extension": ".docx",
                        "mime": "application/vnd.openxmlformats-officedocument.wordprocessingml.document"
                    },
                    {
                        "extension": ".xls",
                        "mime": "application/vnd.ms-excel"
                    },
                    {
                        "extension": ".xlsx",
                        "mime": "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
                    },
                    {
                        "extension": ".ppt",
                        "mime": "application/vnd.ms-powerpoint"
                    },
                    {
                        "extension": ".pptx",
                        "mime": "application/vnd.openxmlformats-officedocument.presentationml.presentation"
                    },
                    {
                        "extension": ".txt",
                        "mime": "text/plain"
                    },
                    {
                        "extension": ".rtf",
                        "mime": "application/rtf"
                    },
                    {
                        "extension": ".odt",
                        "mime": "application/vnd.oasis.opendocument.text"
                    },
                    {
                        "extension": ".ods",
                        "mime": "application/vnd.oasis.opendocument.spreadsheet"
                    },
                    {
                        "extension": ".odp",
                        "mime": "application/vnd.oasis.opendocument.presentation"
                    },
                    {
                        "extension": ".odg",
                        "mime": "application/vnd.oasis.opendocument.graphics"
                    },
                    {
                        "extension": ".odc",
                        "mime": "application/vnd.oasis.opendocument.chart"
                    },
                    {
                        "extension": ".odi",
                        "mime": "application/vnd.oasis.opendocument.image"
                    },
                    {
                        "extension": ".jpg",
                        "mime": "image/jpeg"
                    },
                    {
                        "extension": ".jpeg",
                        "mime": "image/jpeg"
                    },
                    {
                        "extension": ".png",
                        "mime": "image/png"
                    }
                ]
            },
            "user": {
                "role_moderator": "MODERATOR",
                "role_viewer": "VIEWER"
            },
            "whiteboard": {
                "annotations": {
                    "status": {
                        "end": "DRAW_END",
                        "start": "DRAW_START",
                        "update": "DRAW_UPDATE"
                    }
                },
                "toolbar": {
                    "colors": [
                        {
                            "label": "black",
                            "value": "#000000"
                        },
                        {
                            "label": "white",
                            "value": "#ffffff"
                        },
                        {
                            "label": "red",
                            "value": "#ff0000"
                        },
                        {
                            "label": "orange",
                            "value": "#ff8800"
                        },
                        {
                            "label": "eletricLime",
                            "value": "#ccff00"
                        },
                        {
                            "label": "Lime",
                            "value": "#00ff00"
                        },
                        {
                            "label": "Cyan",
                            "value": "#00ffff"
                        },
                        {
                            "label": "dodgerBlue",
                            "value": "#0088ff"
                        },
                        {
                            "label": "blue",
                            "value": "#0000ff"
                        },
                        {
                            "label": "violet",
                            "value": "#8800ff"
                        },
                        {
                            "label": "magenta",
                            "value": "#ff00ff"
                        },
                        {
                            "label": "silver",
                            "value": "#c0c0c0"
                        }
                    ],
                    "font_sizes": [
                        {
                            "value": 36
                        },
                        {
                            "value": 32
                        },
                        {
                            "value": 28
                        },
                        {
                            "value": 24
                        },
                        {
                            "value": 20
                        },
                        {
                            "value": 16
                        }
                    ],
                    "multiUserPenOnly": false,
                    "multiUserTools": [
                        "text",
                        "line",
                        "ellipse",
                        "triangle",
                        "rectangle",
                        "pencil",
                        "hand"
                    ],
                    "presenterTools": [
                        "text",
                        "line",
                        "ellipse",
                        "triangle",
                        "rectangle",
                        "pencil",
                        "hand"
                    ],
                    "thickness": [
                        {
                            "value": 14
                        },
                        {
                            "value": 12
                        },
                        {
                            "value": 10
                        },
                        {
                            "value": 8
                        },
                        {
                            "value": 6
                        },
                        {
                            "value": 4
                        },
                        {
                            "value": 2
                        },
                        {
                            "value": 1
                        }
                    ],
                    "tools": [
                        {
                            "icon": "text_tool",
                            "value": "text"
                        },
                        {
                            "icon": "line_tool",
                            "value": "line"
                        },
                        {
                            "icon": "circle_tool",
                            "value": "ellipse"
                        },
                        {
                            "icon": "triangle_tool",
                            "value": "triangle"
                        },
                        {
                            "icon": "rectangle_tool",
                            "value": "rectangle"
                        },
                        {
                            "icon": "pen_tool",
                            "value": "pencil"
                        },
                        {
                            "icon": "hand",
                            "value": "hand"
                        }
                    ]
                }
            }
        }
    }
}

TASK [n0emis.bigbluebutton : write back new meteor config] *********************
fatal: [<hostname>]: FAILED! => {"msg": "Unexpected templating type error occurred on ({{ meteor  | to_nice_yaml }}): value must be a string"}

PLAY RECAP *********************************************************************
<hostname> : ok=55   changed=0    unreachable=0    failed=1    skipped=34   rescued=0    ignored=0   
```

**Comment by fkohrt** [comment]
> Which version of ansible are you using?

ansible 2.9.10

**Comment by fkohrt** [comment]
I couldn't help myself but to install a fresh Ubuntu Server 16.04 LTS on the target; I also used a fresh Ubuntu Server 20.04 LTS to run Ansible from and now it worked. Well at least the task "write back new meteor config" was working; I also had issue #48 but was able to fix it with the provided [solution](https://github.com/n0emis/ansible-role-bigbluebutton/issues/48#issue-638839681).

**Comment by fkohrt** [comment]
Closing this as it was likely due to some local configuration issues (though I did not investigate further, but used a clean environment instead).

**Comment by anadahz** [comment]
I had the same issue and resolved it by using a newer ansible version (`2.10.1`).
